File: android-java-maven-single/app/src/com/mycompanyname/bananas/TempActivity.java
package com.mycompanyname.bananas;

import android.app.Activity;
import android.widget.TextView;

import com.google.java.contract.Requires;

import com.googlecode.androidannotations.annotations.Click;
import com.googlecode.androidannotations.annotations.EActivity;
import com.googlecode.androidannotations.annotations.ViewById;

@EActivity(R.layout.temp)
public class TempActivity extends Activity {

	@ViewById(R.id.txt) TextView textView;

	@Click(R.id.btn)
	@Requires("textView != null")
	public void onButton() {
		textView.setText("great");
	}
}

File: android-scala-maven-single/app/src/main/scala/TempActivity.scala
package com.mycompanyname.bananas

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class TempActivity extends Activity {

  lazy val textView = findViewById(R.id.txt).asInstanceOf[TextView]

  override def onCreate(savedInstanceState: Bundle) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.temp)
  }

  def onButton(v: View) {
    textView.setText("great")
  }
}
